### PR TITLE
Cohort service

### PR DESCRIPTION
**Pull request by luisgarcc**
This PR implements the first version of the Cohort Service App. The Cohort Service App is a Java cloud application built using Spring that uses the quality-measure-and-cohort-service code to execute CQL libraries against patients in a FHIR server. 

The application is comprised of multiple REST API endpoints that allow users to:

1. Configure a FHIR connection
2. Manage CQL libraries (including all CRUD operations) that specify Clinical Quality Measures
3. Run those CQL libraries against all or subset of patients in the FHIR server

This delivery includes:
- Java code
- Docker image
- Kubernetes manifests
- Helm Chart
- Docs

### CODE REVIEW

File: cohort-service/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<p align="center">
+    <a href="https://github.com/Alvearie">
+        <img src="https://avatars.githubusercontent.com/u/72946463?s=200&v=4" height="100" alt="Alvearie">
+    </a>
+</p>
+
+
+<p align="center">
```
Comment (ibmatc) [inline cohort-service/README.md:8]: Comment for another issue - I think we should be consistent with page decorations. This header and the license/contribution guidelines at the bottom - we should clean up documentation.

File: cohort-service/src/main/java/com/ibm/healthpatterns/app/CohortService.java [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+	/**
+	 * Loads the default FHIR configuration for the FHIR server to whcih the service will connect to query patients.
+	 */
+	private void loadDefaultFHIRConfig() {
+		ObjectMapper mapper = new ObjectMapper();
+		ClassPathResource fhirFile = new ClassPathResource("config/default-ibm-fhir.json");
+		try {
+			fhirConnectionInfo = mapper.readValue(fhirFile.getInputStream(), FhirServerConfig.class);
+		} catch (JsonMappingException e) {
+			System.err.println("The default IBM FHIR connection JSON file does not match the corresponding connection class: " + e.getMessage());
```
Comment (ibmatc) [inline cohort-service/src/main/java/com/ibm/healthpatterns/app/CohortService.java:102]: why System.err/out vs Logger?

File: cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	/**
+	 * 
+	 * @param libraryID the library id
+	 * @param ids the optional list of patient IDs to run the cohort, instead of using all patients in the FHIR server
+	 * @param reverse run the CQL engine with a reversed where only the patients that don't match the cohort are returned
+	 * @return the list of patients
+	 */
+	@GetMapping("/patientIds")
+	public @ResponseBody ResponseEntity<String> getPatientIds(@PathVariable String libraryID, 
+															  @RequestParam(required = false) String ids,
```
Comment (ibmatc) [inline cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java:68]: Formatting looks a little odd here.  Is this just a GitHub review issue?

File: cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	/**
+	 * 
+	 * @param libraryID the library id
+	 * @param ids the optional list of patient IDs to run the cohort, instead of using all patients in the FHIR server
+	 * @param reverse run the CQL engine with a reversed where only the patients that don't match the cohort are returned
+	 * @return the list of patients
+	 */
+	@GetMapping("/patients")
+	public @ResponseBody ResponseEntity<String> getPatients(@PathVariable String libraryID, 
+															@RequestParam(required = false) String ids,
```
Comment (ibmatc) [inline cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java:100]: Same formatting issue

**Review by ibmatc — APPROVED** [review]
LGTM

File: cohort-service/README.md (hunk 1 above)
Comment (luisgarcc) [reply cohort-service/README.md:8]: I liked how this looked and left it, but I agree we need to converge.

File: cohort-service/src/main/java/com/ibm/healthpatterns/app/CohortService.java (hunk 2 above)
Comment (luisgarcc) [reply cohort-service/src/main/java/com/ibm/healthpatterns/app/CohortService.java:102]: I had to spend some more time figuring out how to deal with logging in _SpringBoot_ the right way, which was time I didn't want to lose for this version of the prototype. I eventually ran into how to configure it though, so it's in my (long) list of things to enhance for a future version. 😄

File: cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java (hunk 3 above)
Comment (luisgarcc) [reply cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java:68]: I think so, it looks good in the IDE:

image

File: cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java (hunk 4 above)
Comment (luisgarcc) [reply cohort-service/src/main/java/com/ibm/healthpatterns/rest/controllers/PatientsController.java:100]: Ditto.
image

**Comment by luisgarcc** [comment]
> 
> 
> LGTM

Thanks!

**Pull request merged**